### PR TITLE
Characterp

### PR DESCRIPTION
**Pull request by Izzeri**
Moved `characterp` from C to Rust! This is my first PR ever, so I hope I did everything correctly.

### CODE REVIEW

File: rust_src/src/character.rs [hunk 1]
```diff
@@ -21,3 +21,28 @@ defun!("max-char",
...
        "Return the character of the maximum code.");
+
+/// Is this LispObject a character?
+fn CHARACTERP(obj: LispObject) -> bool {
+    lisp::NATNUMP(obj) && lisp::XFASTINT(obj) <= MaxChar
+}
+
+fn Fcharacterp(obj: LispObject, _ignore: LispObject) -> LispObject {
+    if CHARACTERP(obj) {
+        unsafe { lisp::Qt }
```
Comment (jeandudey) [inline rust_src/src/character.rs]: Use `LispObject::constant_t` instead

File: rust_src/src/character.rs [hunk 2]
```diff
@@ -21,3 +21,28 @@ defun!("max-char",
...
+    } else {
+        lisp::Qnil
```
Comment (jeandudey) [inline rust_src/src/character.rs]: Use `LispObject::constant_nil` instead

File: rust_src/src/character.rs [hunk 3]
```diff
@@ -21,3 +21,28 @@ defun!("max-char",
        0,
        ptr::null(),
        "Return the character of the maximum code.");
+
+/// Is this LispObject a character?
+fn CHARACTERP(obj: LispObject) -> bool {
+    lisp::NATNUMP(obj) && lisp::XFASTINT(obj) <= MaxChar
```
Comment (jeandudey) [inline rust_src/src/character.rs]: I think these deprecated functions could replaced by `LispObject` functions

File: src/character.c [hunk 4]
```diff
@@ -1051,7 +1051,7 @@ syms_of_character (void)
   staticpro (&Vchar_unify_table);
   Vchar_unify_table = Qnil;
 
-  defsubr (&Scharacterp);
+  //  defsubr (&Scharacterp);
```
Comment (jeandudey) [inline src/character.c]: Remove this, we don't need it anymore

**Comment by Wilfred** [comment]
I believe the failing test has been fixed on master, so if you rebase on master, or merge master into your branch, the build should go green too :)

**Comment by Izzeri** [comment]
Hey! Sorry for not replying sooner! I'll fix these when I have time. Busy week at work.

**Comment by jeandudey** [comment]
This was implemented on #170, closing this.

**Pull request closed without merging**